### ISSUE TITLE
[cloud_firestore] Flutter app crashes when using firestore and functions together

### ISSUE THREAD

**Issue by kroikie** [FirebaseExtended/flutterfire#1214]
I was using old version of firestore plugin 0.8 i remember, transactions was not working, then i get the 0.12.7 and now when i use firestore functions app crashes. Like 1 or 2 month ago app was working correctly. I can't figure out the problem.

```
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): Failed to handle method call result
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): java.lang.ClassCastException: java.lang.String cannot be cast to java.util.Map
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$4$1$1.success(CloudFirestorePlugin.java:391)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at io.flutter.plugin.common.MethodChannel$IncomingResultHandler.reply(MethodChannel.java:199)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at io.flutter.embedding.engine.dart.DartMessenger.handlePlatformMessageResponse(DartMessenger.java:114)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessageResponse(FlutterJNI.java:654)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at android.os.MessageQueue.next(MessageQueue.java:326)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at android.os.Looper.loop(Looper.java:160)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at android.app.ActivityThread.main(ActivityThread.java:6669)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/MethodChannel#plugins.flutter.io/cloud_firestore( 9595): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)`

`E/CloudFirestorePlugin( 9595): Timed out waiting for Task
E/CloudFirestorePlugin( 9595): java.util.concurrent.TimeoutException: Timed out waiting for Task
E/CloudFirestorePlugin( 9595): 	at com.google.android.gms.tasks.Tasks.await(Unknown Source:32)
E/CloudFirestorePlugin( 9595): 	at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$4.apply(CloudFirestorePlugin.java:418)
E/CloudFirestorePlugin( 9595): 	at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$4.apply(CloudFirestorePlugin.java:370)
E/CloudFirestorePlugin( 9595): 	at com.google.firebase.firestore.FirebaseFirestore.lambda$runTransaction$1(com.google.firebase:firebase-firestore@@19.0.0:283)
E/CloudFirestorePlugin( 9595): 	at com.google.firebase.firestore.FirebaseFirestore$$Lambda$3.call(Unknown Source:6)
E/CloudFirestorePlugin( 9595): 	at com.google.android.gms.tasks.zzv.run(Unknown Source:2)
E/CloudFirestorePlugin( 9595): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/CloudFirestorePlugin( 9595): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/CloudFirestorePlugin( 9595): 	at java.lang.Thread.run(Thread.java:764)`

`D/AndroidRuntime( 9595): Shutting down VM
E/AndroidRuntime( 9595): FATAL EXCEPTION: main
E/AndroidRuntime( 9595): Process: com.acmhacettepe.acmhacettepe, PID: 9595
E/AndroidRuntime( 9595): java.lang.IllegalStateException: Reply already submitted
E/AndroidRuntime( 9595): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:135)
E/AndroidRuntime( 9595): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.error(MethodChannel.java:230)
E/AndroidRuntime( 9595): 	at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$3.onComplete(CloudFirestorePlugin.java:436)
E/AndroidRuntime( 9595): 	at com.google.android.gms.tasks.zzj.run(Unknown Source:4)
E/AndroidRuntime( 9595): 	at android.os.Handler.handleCallback(Handler.java:873)
E/AndroidRuntime( 9595): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 9595): 	at android.os.Looper.loop(Looper.java:193)
E/AndroidRuntime( 9595): 	at android.app.ActivityThread.main(ActivityThread.java:6669)
E/AndroidRuntime( 9595): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 9595): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/AndroidRuntime( 9595): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
I/Process ( 9595): Sending signal. PID: 9595 SIG: 9
Lost connection to device.
```

**Comment by kroikie** [comment]
@emrekosen

The issue at https://github.com/flutter/flutter/issues/36189 has been closed and moved here. Future collaboration on this issue will be done here.

**Comment by tewshi** [comment]
Just adding Cloud Firestore crashes my app, Android and iOS (in debug mode) without any error on console.

**Comment by Adejair** [comment]
i have same error, any answer?

**Comment by tewshi** [comment]
I've been using iPhone running on iOS 13 to test, and nothing logs on the console, but when I ran the app on Android, I noticed this error:
```
java.lang.RuntimeException: Unable to get provider com.google.firebase.provider.FirebaseInitProvider: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.provider.FirebaseInitProvider"...
```
this came up after enabling `Multidex`

**Comment by kroikie** [comment]
@tewshi Are you using other FlutterFire plugins? Would you also mind sharing the dependency portion of your pubspec?

**Comment by yvzoncu** [comment]
cupertino_icons: ^0.1.2
  firebase_core: ^0.4.0+9
  cloud_firestore: ^0.12.9+5
  firebase_auth: ^0.14.0+5
  modal_progress_hud: ^0.1.3
  morpheus: ^1.2.2+1
  animated_text_kit: ^1.3.1
  random_string: ^1.1.0
  scoped_model: ^1.0.1
  get_it: ^3.0.0+1
  intl: ^0.16.0
  titled_navigation_bar: ^3.6.0
  flutter_launcher_icons: ^0.7.3
  meta: ^1.1.7

here u go

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by kroikie** [comment]
@yvzoncu @tewshi this issue is related to using `cloud_firestore` and `cloud_functions` together. Seems like your issues are unrelated to this. Please file separate issues so the can be addressed individually.

The title of this issue is a bit of a catch all I'll update it.

**Comment by kroikie** [comment]
@tewshi thanks for sharing that. Could you confirm that iOS <=12 and Android <= 7 have this issue?

**Comment by youssefhad** [comment]
same issue , Android 9

**Comment by pinkfish** [comment]
I am seeing this with the latest plugin and with transactions.  I aded in a runTransation call and this is the error I get.

**Comment by Salakar** [comment]
Hey, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
